File: old_trunk/cost/iona/orbacus/cpp/poa/TestClient.cpp
// **********************************************************************
//
// Copyright (c) 2001
// Object Oriented Concepts, Inc.
// Billerica, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

#include <includes.h>

#include <Test.h>

#include <fstream>

using namespace CORBA;

int
main(int argc, char* argv[], char*[])
{
    int status = EXIT_SUCCESS;
    ORB_var orb;

    try
    {
        orb = ORB_init(argc, argv);

        const char* refFile = "Test.ref";
        ifstream in; // Must use open(name), not ifstream in(name) (VC++ bug)
        in.open(refFile);
        if(in.fail())
        {
            OB_ERROR(argv[0] << ": can't open `" << refFile << "'");
            return EXIT_FAILURE;
        }

        char s[1000];
        in >> s;
	in.close();

        Object_var obj = orb -> string_to_object(s);
        if(CORBA::is_nil(obj))
        {
            OB_ERROR(argv[0] << ": cannot read IOR from Test.ref");
            return EXIT_FAILURE;
        }
        TestServer_var server = TestServer::_narrow(obj);
        TEST(!is_nil(server));

        TestInfoSeq_var info = server -> get_info();

        for(ULong i = 0 ; i < info -> length() ; i++)
        {
            try
            {
                info[i].obj -> aMethod();
                TEST(strlen(info[i].except_id) == 0);
            }
            catch(const SystemException& ex)
            {
#ifndef HAVE_NO_CORBA_2_4
                if(strcmp(info[i].except_id, ex._rep_id()) != 0)
                {
                    OB_ERROR("TestClient: Invocation on object #" << i
			     << " raised an unexpected exception");
                    OB_ERROR("Expected " << info[i].except_id
			     << " but caught " << ex._rep_id());
                    TEST(false);
                }
#else
		TEST(false);
#endif
            }
        }

	server -> deactivate();
    }
    catch(const Exception& ex)
    {
#ifdef HAVE_EXCEPTION_INSERTERS
	OB_ERROR(ex);
#endif
        status = EXIT_FAILURE;
    }

    if(!is_nil(orb))
    {
	try
	{
	    orb -> destroy();
	}
	catch(const Exception& ex)
	{
#ifdef HAVE_EXCEPTION_INSERTERS
	    OB_ERROR(ex);
#endif
	    status = EXIT_FAILURE;
	}
    }

    return status;
}

File: cost/iona/orbacus/java/poa/TestClient.java
// **********************************************************************
//
// Copyright (c) 2001
// Object Oriented Concepts, Inc.
// Billerica, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

package test.poa;

import java.io.*;
import org.omg.CORBA.*;

public class TestClient extends test.common.TestBase
{
    public static void
    main(String[] args)
    {
	java.util.Properties props = System.getProperties();

        ORB orb = null;

	try
	{
            //
            // Create ORB
            //
            orb = ORB.init(args, props);

	    String impl = null;

	    //
	    // Get server
	    //
	    try
	    {
		String refFile = "Test.ref";
		FileInputStream file = new FileInputStream(refFile);
		BufferedReader in =
		    new BufferedReader(new InputStreamReader(file));
		impl = in.readLine();
		file.close();
	    }
	    catch(IOException ex)
	    {
		System.err.println("Can't read from `" +
				   ex.getMessage() + "'");
		System.exit(1);
	    }

	    org.omg.CORBA.Object obj = orb.string_to_object(impl);
	    TestServer server = TestServerHelper.narrow(obj);
	    TEST(server != null);

	    TestInfo[] info = server.get_info();
	    for(int i = 0 ; i < info.length ; i++)
	    {
		try
		{
		    info[i].obj.aMethod();
		    TEST(info[i].except_id.length() == 0);
		}
		catch(SystemException ex)
		{
                    String id = test.common.Util.getExceptionId(ex);
                    if(!id.equals(info[i].except_id))
                    {
                        System.err.println("TestClient: Invocation on " +
                                           "object #" + i + " raised an " +
                                           "unexpected exception");
                        System.err.println("Expected " + info[i].except_id +
                                           " but caught " + id);
                        TEST(false);
                    }
		}
	    }

	    server.deactivate();
	}	    
	catch(SystemException ex)
	{
	    ex.printStackTrace();
	    System.exit(1);
	}

        if(orb != null)
        {
            try
            {
                orb.destroy();
            }
            catch(SystemException ex)
            {
                ex.printStackTrace();
                System.exit(1);
            }
        }

        System.exit(0);
    }
}
